jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +64,43 @@ def check_referer(self):
             return False
         return True
 
+    def get_scope_filter(self, req_scope):
```

Comment:
I think scope_filter probably wants to produce a callable instead of a set. That way, it doesn't need to resolve all possible answers before checking values.